### ISSUE TITLE
Bind the GPU gesvd

### ISSUE THREAD

**Issue by Roger-luo** [Roger-luo/MAGMA.jl#18]

**Comment by Roger-luo** [comment]
This is sort of resolved, but we will need some frontend wrapper later.

**Comment by Roger-luo** [comment]
I think we now just need to define a generic function (but do not overload the `svd` in LinearAlgebra` and dispatch it to magma `gesdd` in this repo as the upper most interface.

**Comment by JerryChen97** [comment]
Yes one month ago I read their code for the gpu version of magma_gesdd in pytorch and followed but only did this stuff in the test file test/dense/svd.jl as we can see.
I will open a new PR and push a dispatched src.
